airbyte-lib/tests/integration_tests/test_source_faker_integration.py

```diff
@@ -0,0 +1,216 @@
+    assert catalog.streams[0].primary_key
+    assert catalog.streams[1].primary_key
+
+    read_result = source_faker_seed_a.read(duckdb_cache, write_strategy="append")
+    assert read_result.cache._get_primary_keys("products") == ["id"]
+    assert read_result.cache._get_primary_keys("purchases") == ["id"]
+
+
+def test_replace_strategy(
+    source_faker_seed_a: ab.Source,
+    all_cache_types: ab.DuckDBCache,
+) -> None:
+    """Test that the append strategy works as expected."""
+    for cache in all_cache_types: # Function-scoped fixtures can't be used in parametrized().
+        for _ in range(2):
+            result = source_faker_seed_a.read(
+                cache, write_strategy="replace", force_full_refresh=True
+            )
+            assert len(result.cache.streams) == 2
+            assert len(list(result.cache.streams["products"])) == NUM_PRODUCTS
+            assert len(list(result.cache.streams["purchases"])) == FAKER_SCALE_A
+
+
+def test_append_strategy(
+    source_faker_seed_a: ab.Source,
+    all_cache_types: ab.DuckDBCache,
+) -> None:
+    """Test that the append strategy works as expected."""
+    for cache in all_cache_types: # Function-scoped fixtures can't be used in parametrized().
+        for iteration in range(1, 3):
+            result = source_faker_seed_a.read(cache, write_strategy="append")
+            assert len(result.cache.streams) == 2
+            assert len(list(result.cache.streams["products"])) == NUM_PRODUCTS * iteration
+            assert len(list(result.cache.streams["purchases"])) == FAKER_SCALE_A * iteration
+
+
+@pytest.mark.parametrize("strategy", ["merge", "auto"])
+def test_merge_strategy(
+    strategy: str,
+    source_faker_seed_a: ab.Source,
+    source_faker_seed_b: ab.Source,
+    all_cache_types: ab.DuckDBCache,
+) -> None:
+    """Test that the merge strategy works as expected.
+
+    Since all streams have primary keys, we should expect the auto strategy to be identical to the
+    merge strategy.
+    """
+    for cache in all_cache_types: # Function-scoped fixtures can't be used in parametrized().
+        # First run, seed A (counts should match the scale or the product count)
+        result = source_faker_seed_a.read(cache, write_strategy=strategy)
+        assert len(result.cache.streams) == 2
+        assert len(list(result.cache.streams["products"])) == NUM_PRODUCTS
+        assert len(list(result.cache.streams["purchases"])) == FAKER_SCALE_A
+
+        # Second run, also seed A (should have same exact data, no change in counts)
+        result = source_faker_seed_a.read(cache, write_strategy=strategy)
+        assert len(list(result.cache.streams["products"])) == NUM_PRODUCTS
+        assert len(list(result.cache.streams["purchases"])) == FAKER_SCALE_A
+
+        # Third run, seed B - should increase record count to the scale of B, which is greater than A.
+        # TODO: See if we can reliably predict the exact number of records, since we use fixed seeds.
```

Comment:
from what I can tell we can rely on this (especially now with the pinned version of faker)